agent/kafka_handler.go

```diff
@@ -17,6 +29,130 @@ type kafkaHandler struct {
 	ctx   kafkaserver2.ConnectionContext
 }
 
+func extractErrorCode(err error) common.ErrCode {
+	var tektiteErr common.TektiteError
+	if errors.As(err, &tektiteErr) {
+		return tektiteErr.Code
+	}
+	return common.UnknownError
```

Comment:
No need to create a common.UnknownError - just return the TektiteError and UnknownServerError otherwise

Reply:
makes sense. Done